### ISSUE TITLE
Shopping List Scaling

### ISSUE THREAD

**Issue by vabene1111** [TandoorRecipes/recipes#453]
The new servings system made the ingredient calculation not based on 1 but the servings of the recipe.

This has been implemented in the recipe view and meal plan and also to a certain degree in the shopping list but for some reason the base servings have been ignored here

- [ ] find the best way to store the recipe base servings in the shopping list
- [ ] use base servings in the amount calculation (item amount / base servings * current servings)

**Comment by LennyPenny** [comment]
So I'm a bit confused by the wording of this ticket so I want to make sure that it covers the bug I am experiencing: 
The number of servings selected in the meal plan is not followed when the recipes are added to the shopping list from the meal plan. The shopping list displays ingredients needed for default amount of servings from the recipe. 

So for example if I have a recipe with a default number of servings 4, then make a meal plan with the recipe in it for two servings and then add that to a shopping list, the ingredients needed for 4 servings are added to the shopping list.

**Comment by vabene1111** [comment]
yes that is correct. Basically there was a scaling system before servings were added (which just multiplied the recipe ingredients). After adding servings people wanted to have scaling by servings (which is definitely a good thing) but i made a logical error when implementing that part for the shopping list so it basically always uses the old system.

Since the fix is a little more complicated it is like this for now, i hope to find time for this soonish

**Comment by lucker999** [comment]
If I understand correctly, at this moment I can not add a recipe with saved 8 servings to mealplan to get, say, shopping list for 6 servings? If this is not true, how can I do that?

I have stored most of the recipes with default 8 servings each. Now I want to use the same recipes to compose a shopping list (or meal plan) for 6 servings in each recipes. Is it possible? Or we should wait for the time being till the logic will be corrected?

BTW, thank you a lot for such a useful tool. It's awesome even with as little functionality as it has at the moment. I really appreciate what you are creating!

**Comment by lucker999** [comment]
@vabene1111 First of all, I apologize for my vague wording about little functionality. I had absolutely no intentions to belittle this project, which is awesome in many dimensions: idea, code quality, docs, your responsiveness as a developer and many more. You are certainly right, as a recipe tool for a person or a small household it has a lot useful features, which I personally appreciate very much indeed. All I meant is that there are some other possible use cases I can imagine right of the bat and probably even more worth pursuing (if it is of any interest) since the idea and this project as its implementation has a good potential. For those use cases (which probably reside only in my imagination, - not tested, but for one such use case I have implemented Tandoor) the existing features are not enough. I can provide more detail if any interest from your side, but, alas, I do not know how to send a direct message on GitHub.

As for feature examples, I would appreciate synonyms for ingredient, especially useful with import from websites or even rules to convert some ingredients by mask or with wildcard during import. For example, ingredients during import from some websites include quantity, e.g. 'milk - 1 glass' which appears in the Name field. Another problem is different wording of the same ingredient, e.g. "Chopped onion" and "Onion, chopped", which is basically the same, but creates two lines in shopping list. But it is just one example. For a person, who import a recipe occasionally once in a month this is probably not a big deal.

Another example is a unit conversion for an ingredient or type of ingredients, e.g. for Öl 1 Esslöffel = 15 ml = 0.015 L, but for salt 1 tsp = 7 g. More important, for each use case the conversion feature should be slightly different (settings in options, perhaps) since it would involve different workflows. And aggregating units in shopping list, e.g. instead 1750 ml of milk it should put a 2x1L Milk or something like that. 

Also thank you for the answer. After you kindly explained this way to adjust shopping list for changed servings, now it seems obvious to me.

**Comment by vabene1111** [comment]
Thank you very much for the detailed feedback.

There was no reason to apologize, I did not feel bad (for lack of a better word) I just wanted to try to better understand how to improve. Feedback is most times related to a lack of certain features relevant for certain people or use cases (as in your case) or sometimes to a lack of good documentation or a bad UI / UX design.

In your case I think most of the things you propose are already mentioned in some form in a special issue. If you **want** feels free to search them (I am on my phone and can't find them right now) and comment on them if you have anything to add (which I feel like you made some good points, especially since a rule system was just in discussion the other day for food aliases).

Other than that feel free to contact me at info@tandoor.dev if you want to speak to me in private. As I saw that you used German units feel free to use German or English whichever you prefer.
